src/components/TodoContainer.js

```diff
@@ -0,0 +1,216 @@
+
+      // Handle errors during adding a todo
+    } catch (error) {
+      console.log("Error fetching data:", error.message);
+      return null;
+    }
+  };
+
+  //Delete items method DELETE
+  const removeTodo = async (id) => {
+    const options = {
+      method: "DELETE",
+      headers: {
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        "Content-Type": "application/json",
+      },
+    };
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        throw new Error(`Error has occurred: ${response.status}`);
+      }
+
+      fetchData(); //deleting items from todo list
+
+      return await response.json();
+    } catch (error) {
+      console.log(error.message);
+      return null;
+    }
+  };
+
+  //Checkbox  Method PATCH
+  const updateData = useCallback(async (id, newValue) => {
+    const options = {
+      method: "PATCH",
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Bearer ${process.env.REACT_APP_AIRTABLE_API_TOKEN}`,
+        // "Authorization": `Bearer ${process.env.REACT_APP_AIRTABLE_API_KEY}`
+      },
+      body: JSON.stringify({
+        fields: {
+          CheckBox: newValue.CheckBox,
+        },
+      }),
+    };
+
+    console.log(`Updating record with id: ${id}, newValue: ${newValue}`);
+    try {
+      const response = await fetch(`${AirtableUrl}/${id}`, options);
+      if (!response.ok) {
+        const message = `Error has ocurred: ${response.status}`;
+        throw new Error(message);
+      }
+      const data = await response.json();
+      // console.log(data); // Log the response data to the console
+      // console.log(`Updating record with id: ${id}, newValue:`, newValue);
+      return data;
+    } catch (error) {
+      console.log("Error updating data: ", error);
+    }
+  }, []);
+
+  // Sorting by JavaScript
+  //   The sorting order is determined by the sortOrder state variable.
+  //   After sorting, the sorted list is set as the new state for todoList.
+  //
+
+  const sortData = useCallback(() => {
+    setTodoList((prevTodoList) => {
+      const sorted = [...prevTodoList].sort((objectA, objectB) => {
+        const titleA = objectA.title ? objectA.title.toUpperCase() : "";
+        const titleB = objectB.title ? objectB.title.toUpperCase() : "";
+
+        if (titleA < titleB) {
+          return sortOrder === "asc" ? -1 : 1;
+        } else if (titleA > titleB) {
+          return sortOrder === "asc" ? 1 : -1;
+        } else {
+          return 0;
+        }
+      });
+
+      // Only return the sorted array if it's different from the current one otherwise got an infinite loop
+      if (JSON.stringify(sorted) === JSON.stringify(prevTodoList)) {
+        return prevTodoList;
+      } else {
+        return sorted;
+      }
+    });
+  }, [sortOrder]);
+
+  useEffect(() => {
+    sortData();
+  }, [todoList, sortOrder, sortData]);
```

Comment:
Requires change:
See feedback above (toggling the `sortOrder`) in regards to how this can be changed.